go/vt/vttablet/tabletmanager/vreplication/vreplicator.go

```diff
@@ -305,6 +305,13 @@ func (vr *vreplicator) replicate(ctx context.Context) error {
 					return err
 				}
 			} else {
+				if vr.state != binlogdatapb.VReplicationWorkflowState_Copying {
+					if err := vr.setState(binlogdatapb.VReplicationWorkflowState_Copying, ""); err != nil {
+						vr.stats.ErrorCounts.Add([]string{"Copy"}, 1)
+						return err
+					}
+					vr.insertLog(LogCopyRestart, fmt.Sprintf("Copy phase restarted for %d table(s)", numTablesToCopy))
+				}
```

Comment:
Question: this change seems unrelated to this PR specifically, is that correct? Or is this PR creating the only scenario where this change is needed?

Reply:
Yeah, we need it here. When we are adding reference tables and inserting rows in `_vt.copy_state`, streams will jump back to `Copying` phase, this is when we need to log `Copy phase restarted`.